### PR TITLE
Bind deputies to districts

### PR DESCRIPTION
**Pull request by illusionOfParadise**
https://app.clubhouse.io/enmarche/story/4328/etq-adh%C3%A9rent-avec-le-r%C3%B4le-depute-j-ai-une-managed-area-qui-correspond-%C3%A0-ma-circonscription-d%C3%A9partement

### CODE REVIEW

File: src/Command/BindDeputiesToDistrictsCommand.php [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+            return 1;
+        }
+
+        $this->io->text('Start binding deputies to districts');
+
+        $deputies = $this->decoder->decode(file_get_contents($file), 'csv', [CsvEncoder::DELIMITER_KEY => ';']);
+        $this->io->title(sprintf('Starting bind deputies to districts: %s deputies are about to be binded.', count($deputies)));
+
+        $i = 0;
```
Comment (ottaviano) [inline src/Command/BindDeputiesToDistrictsCommand.php]: You don't really need this variable.

File: src/Command/BindDeputiesToDistrictsCommand.php [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    {
+        $this->em = $em;
+
+        parent::__construct();
+    }
+
+    protected function configure()
+    {
+        $this->addArgument('file', InputArgument::REQUIRED, 'CSV file of deputies with their managed districts to load');
+        $this->setDescription('Bind deputies to districts. Attention! If district has a deputy it will be changed without warning.');
```
Comment (ottaviano) [inline src/Command/BindDeputiesToDistrictsCommand.php]: you can chain the calls 
```
$this
  ->addArgument()
  ->setDescription()
;
```

File: src/Command/BindDeputiesToDistrictsCommand.php [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+                $this->io->error(sprintf("Adherent with uuid '%s' does not exist.", $deputy['uuid']));
+                continue;
+            }
+
+            $district = $this->em->getRepository(District::class)->findOneBy([
+                'departmentCode' => sprintf('%05d', $deputy['code_dpt']),
+                'number' => (int) $deputy['num_circo'],
+            ]);
+            if (!$district) {
+                $this->io->error(sprintf("District with ID '%s' and department code '%s' does not exist.", $deputy['num_circo'], $deputy['code_dpt']));
```
Comment (ottaviano) [inline src/Command/BindDeputiesToDistrictsCommand.php]: maybe add for which adherent uuid we cannot find a district?

File: src/Command/BindDeputiesToDistrictsCommand.php (hunk 1 above)
Comment (illusionOfParadise) [reply src/Command/BindDeputiesToDistrictsCommand.php]: I use it here https://github.com/EnMarche/en-marche.fr/pull/2512/files#diff-ae51f37302ff7c267728b0258b517d83R93

File: src/Command/BindDeputiesToDistrictsCommand.php [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+        $this->io->text('Start binding deputies to districts');
+
+        $deputies = $this->decoder->decode(file_get_contents($file), 'csv', [CsvEncoder::DELIMITER_KEY => ';']);
+        $this->io->title(sprintf('Starting bind deputies to districts: %s deputies are about to be binded.', count($deputies)));
+
+        $i = 0;
+        foreach ($deputies as $deputy) {
+            $adherent = $this->em->getRepository(Adherent::class)->findOneBy(['uuid' => $deputy['uuid']]);
+            if (!$adherent) {
+                $this->io->error(sprintf("Adherent with uuid '%s' does not exist.", $deputy['uuid']));
```
[the remark this answers is not in the document]
Comment (illusionOfParadise) [reply src/Command/BindDeputiesToDistrictsCommand.php]: Yes, it's a one shot command.

Normally all uuid in the csv file should be correct.

File: src/Command/BindDeputiesToDistrictsCommand.php [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+                continue;
+            }
+
+            $district = $this->em->getRepository(District::class)->findOneBy([
+                'departmentCode' => sprintf('%05d', $deputy['code_dpt']),
+                'number' => (int) $deputy['num_circo'],
+            ]);
+            if (!$district) {
+                $this->io->error(sprintf(
```
[the remark this answers is not in the document]
Comment (illusionOfParadise) [reply src/Command/BindDeputiesToDistrictsCommand.php]: The districts should exist before running the command but have no associated adherent (deputy).

**Pull request merged**